mempool/src/core_mempool/transaction_store.rs

```diff
@@ -343,27 +341,15 @@ impl TransactionStore {
         (batch, last_timeline_id)
     }
 
-    /// Returns transactions with timeline ID in `timeline_ids`
-    /// as list of (timeline_id, transaction)
-    pub(crate) fn filter_read_timeline(
-        &mut self,
-        timeline_ids: Vec<u64>,
-    ) -> Vec<(u64, SignedTransaction)> {
-        timeline_ids
-            .into_iter()
-            .filter_map(|timeline_id| {
-                if let Some((address, sequence_number)) =
-                    self.timeline_index.get_timeline_entry(timeline_id)
-                {
-                    if let Some(txn) = self
-                        .transactions
-                        .get_mut(&address)
-                        .and_then(|txns| txns.get(&sequence_number))
-                    {
-                        return Some((timeline_id, txn.txn.clone()));
-                    }
-                }
-                None
+    pub(crate) fn timeline_range(&mut self, start_id: u64, end_id: u64) -> Vec<SignedTransaction> {
+        self.timeline_index
+            .timeline_range(start_id, end_id)
+            .iter()
+            .filter_map(|(account, sequence_number)| {
+                self.transactions
```

Comment:
Is it possible that there is a race and the transaction is gone from self.transactions here?